lib/child_process.js

```diff
@@ -277,7 +271,7 @@ exports.execFile = function(file /*, args, options, callback*/) {
         if (!encoding)
           _stderr.push(chunk);
         else
-          _stderr += chunk;
+          _stderr += chunk.toString(encoding);
```

Comment:
On second thought, does it make sense to emit on byte count and not character count if encoding is set? Only because the string passed to `'data'` can vary noticeably if reading in utf8 characters.